Lib/doctest.py

```diff
@@ -221,7 +228,7 @@ def _load_testfile(filename, package, module_relative, encoding):
                 file_contents = file_contents.decode(encoding)
                 # get_data() opens files as 'rb', so one must do the equivalent
                 # conversion as universal newlines would do.
-                return file_contents.replace(os.linesep, '\n'), filename
+                return _newline_convert(file_contents), filename
```

Comment:
Instead of trying to replicate the behavior of universal new lines, perhaps we can combine this and the `decode` routine above with something like:

```python
>>> file_contents = b"line1\r\nline2\nline3\rline4"
>>> io.TextIOWrapper(io.BytesIO(file_contents), encoding='utf8').read()
'line1\nline2\nline3\nline4'
```

Reply:
I don't know enough about the internals of TextIOWrapper and BytesIO to know if this would risk breaking anything else. My intent was to make the minimal possible fix to the existing code, and since the existing code used the string replace operation, I just fixed that up to do the correct replacements.

I also don't know if the read operation on TextIOWrapper, or the BytesIO constructor, might throw different exceptions on malformed data than the simple decode and string replace operations would.

Reply:
The concern about possibly changing exceptions is enough for me to want to stick with the simple fix here.